### ISSUE TITLE
RedisAdapter not assignable to AdapterConstructor

### ISSUE THREAD

**Issue by MaximoLiberata** [socketio/socket.io-redis-adapter#482]
```
import { Server as ServerIO } from 'socket.io';
import { createClient as redisCreateClient } from 'redis';
import { createAdapter as redisCreateApter } from '@socket.io/redis-adapter';


const pubClient = redisCreateClient({ url: envConfig.REDIS_HOST });
const subClient = pubClient.duplicate();

const data = await Promise.allSettled([
    pubClient.connect(),
    subClient.connect().then(() => {
        server.#io.adapter(redisCreateApter(pubClient, subClient));
    })
]);

for (const job of data) {

    if (job.status === 'rejected') {
        throw new Error(`Error connect to redis database ${job.reason}`);
    }

}

```

&nbsp;

**Package versions**
&nbsp;
"@socket.io/redis-adapter": "8.0.0",
"redis": "4.5.1",
"socket.io": "4.5.4",
"typescript": "4.9.4"

&nbsp;
&nbsp;

According to TS RedisAdapter not is compatible or assignable to AdapterConstructor.

```
Argument of type '(nsp: any) => RedisAdapter' is not assignable to parameter of type 'AdapterConstructor'.
  Type '(nsp: any) => RedisAdapter' is not assignable to type '(nsp: Namespace<DefaultEventsMap, DefaultEventsMap, DefaultEventsMap, any>) => Adapter'.
    Type 'RedisAdapter' is not assignable to type 'Adapter'.
      Types have separate declarations of a private property 'encoder'.
```

**Comment by Athlon1600** [comment]
Had the same issue... tracked it down to this line which introduced that `encoder` property that TypeScript complains about...

https://github.com/socketio/socket.io-adapter/blob/4e3991f740808afc91eb7b12b8845595cc3a158d/lib/index.ts#L233-L250

It wasn't there back in 2.4.0:  
https://github.com/socketio/socket.io-adapter/compare/2.4.0...2.5.0#diff-d5a4b3a0309f2337144861084c1014523cced561eb722cb9684c63f5e41e0567R231

so temporary solution here I guess is downgrading back to 2.4.0...  
```
npm remove socket.io-adapter@2.5.1
npm install socket.io-adapter@2.4.0
```

Old version compiles fine:

![](https://i.imgur.com/hxfe8ZC.png)

**Comment by MaximoLiberata** [comment]
For now, I down to ```socket.io-adapter@2.4.0```. The issue is with ^2.5.0 version as commented before.

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io-redis-adapter/commit/7aecf01c361602f17ee3759da1ceeabb460d4519, included in version [8.0.1](https://github.com/socketio/socket.io-redis-adapter/releases/tag/8.0.1).
